go-controller/pkg/node/management-port-smartnic.go

```diff
@@ -0,0 +1,170 @@
+package node
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"k8s.io/klog/v2"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+type managementPortSmartNIC struct {
+	nodeName    string
+	hostSubnets []*net.IPNet
+	vfRepName   string
+}
+
+func (mp *managementPortSmartNIC) Create(nodeAnnotator kube.Annotator, waiter *startupWaiter) (*managementPortConfig, error) {
+	// Get management port representor name
+	link, err := util.GetNetLinkOps().LinkByName(mp.vfRepName)
+	if err != nil {
+		// It may fail in case this is not the first run after reboot and management port has already been renamed.
+		link, err = util.GetNetLinkOps().LinkByName(types.K8sMgmtIntfName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get link device for %s. %v", mp.vfRepName, err)
+		}
+	}
+
+	// configure management port: rename, set MTU and set link up and connect representor port to br-int
+	klog.Infof("Create management port smart-nic: %s", link.Attrs().Name)
+	setName := link.Attrs().Name != types.K8sMgmtIntfName
+	setMTU := link.Attrs().MTU != config.Default.MTU
+
+	if setName || setMTU {
+		if err = util.GetNetLinkOps().LinkSetDown(link); err != nil {
+			return nil, fmt.Errorf("failed to set link down for device %s. %v", mp.vfRepName, err)
+		}
+
+		if setName {
+			if err = util.GetNetLinkOps().LinkSetName(link, types.K8sMgmtIntfName); err != nil {
+				// NOTE(adrianc): rename may fail with "file exists" in case an interface is already named
+				// ovn-k8s-mp0, this may happen if mgmt-port-netdev changes during deployment. ATM we are
+				// not handling it.
+				// TODO: handle mgmt-port-netdev change.
+				return nil, fmt.Errorf("failed to set link name for device %s. %v", mp.vfRepName, err)
+			}
+		}
+
+		if setMTU {
+			if err = util.GetNetLinkOps().LinkSetMTU(link, config.Default.MTU); err != nil {
+				return nil, fmt.Errorf("failed to set link MTU for device %s. %v", link.Attrs().Name, err)
+			}
+		}
+	}
+
+	if err = util.GetNetLinkOps().LinkSetUp(link); err != nil {
+		return nil, fmt.Errorf("failed to set link up for device %s. %v", link.Attrs().Name, err)
+	}
+
+	// Plug management port VF representor to OVS.
+	stdout, stderr, err := util.RunOVSVsctl(
+		"--", "--may-exist", "add-port", "br-int", types.K8sMgmtIntfName,
+		"--", "set", "interface", types.K8sMgmtIntfName,
+		"external-ids:iface-id="+types.K8sPrefix+mp.nodeName)
```

Comment:
Indeed conceptually they are the same, but not exactly.

1. it does not deal (and does not need to deal) with legacy interface names
2. it is not an internal port
3. MTU is set on the VF representor and not via ovs

here is the bit of code for the non-smart-nic version:

```
	legacyMgmtIntfName := util.GetLegacyK8sMgmtIntfName(mp.nodeName)
	stdout, stderr, err := util.RunOVSVsctl(
		"--", "--if-exists", "del-port", "br-int", legacyMgmtIntfName,
		"--", "--may-exist", "add-port", "br-int", types.K8sMgmtIntfName,
		"--", "set", "interface", types.K8sMgmtIntfName,
		"type=internal", "mtu_request="+fmt.Sprintf("%d", config.Default.MTU),
		"external-ids:iface-id="+types.K8sPrefix+mp.nodeName)
```

we can probably make a generic enough method for the "add-port" part to serve both cases but i dont think it would benefit code readability much, so id prefer to leave it as is for now.

Reply:
hm... maybe there should just be more comments clarifying the ways that "smart-nic"+"smart-nic-host" is still different from "full"?

The main point of my comment is that I worry that someone will make a change to the "full" implementation and not notice that a very similar change may be needed in the "smart-nic" or "smart-nic-host" implementation...

Reply:
longer term i think the best way is to have some CI running on this hardware.
Im also working on design doc which depicts the changes which im happy to then upload to `docs`.

Reviewers would hopefully be aware of this to reduce chance of breakage.

I can add some comments explaining these implementations with another comment in `management-port.go` saying something of the sort: "if you change stuff here, changes may also be needed in `management-port-smartnic.go` as well"

WDYT ?